cime_config/config_grids.xml

```diff
@@ -2784,6 +2794,12 @@
 
     <!-- ROF (river) grids-->
 
+    <domain name="r2">
```

Comment:
@rljacob Comparing the `dlnd_in` for ` ERS.r05_r05.RMOSGPCC.cori-knl_intel.mosart-gpcc_1972` test, it appears that there is no entry for land domain file in `dlnd_in` for `--res r2_r2 --compset RMOSGPCC`. I believe you need to specify a grid file here for `atm|lnd`